src/tools/clippy/clippy_utils/src/ast_utils/mod.rs

```diff
@@ -444,6 +444,7 @@ pub fn eq_item_kind(l: &ItemKind, r: &ItemKind) -> bool {
         },
         (
             Trait(box ast::Trait {
+                impl_restriction: _, // does not affect item kind
```

Comment:
If I have `impl(self) impl Trait {}` and `impl(super) impl Trait {}` they are different right?

We should probably change this method to compare them.

Reply:
It would be `impl(self) trait Foo {}` (the trait definition rather than implementation). I don't think this would change anything, just as visibility doesn't. It changes what downstream users (incl. other modules) can do but nothing else.